### ISSUE TITLE
We need to set affine=true for nn.SpatialBatchNormalization layer

### ISSUE THREAD

**Issue by Jongchan** [cvpa3/dl-practice#9]
I've found an example using batch normalization [here](https://github.com/eladhoffer/ImageNet-Training/blob/master/Models/AlexNet_BN.lua),
and there states
```
local features = nn.Sequential()
features:add(SpatialConvolution(3,64,11,11,4,4,2,2))       -- 224 -> 55
features:add(SpatialMaxPooling(3,3,2,2))                   -- 55 ->  27
features:add(cudnn.ReLU(true))
features:add(nn.SpatialBatchNormalization(64,nil,nil,false))
```
The last option for SpatialBatchNormalization, which is **affine** is set as false.

However, I had **"Only CUDA Tensors  are supported for cudnn.BatchNormalization layer"** error. 
I guess the cause of this error is **the conversion to cudnn**.

I initialized the model with **nn** module, but later in the skeleton code, where the backend option is set as 'cudnn', the model is converted to **cudnn** type. However, cudnn BatchNormalization only accepts affine=true. So, without initialized

There is a workaround [here](https://github.com/soumith/cudnn.torch/issues/219), but the simplest solution is to **set affine option as true**, as long as you want to use cudnn.
